tests/PHPStan/Analyser/nsrt/uppercase-string-parse.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace UppercaseStringParseStr;
+
+use function PHPStan\Testing\assertType;
+
+class Foo
+{
+
+	/**
+	 * @param uppercase-string $uppercase
+	 */
+	public function parse(string $uppercase, string $string): void
+	{
+		$a = [];
+		parse_str($uppercase, $a);
+
+		if (array_key_exists('foo', $a)) {
+			$value = $a['foo'];
+			if (\is_string($value)) {
+				assertType('lowercase-string|uppercase-string', $value);
```

Comment:
The fact tests are green with `lowercase-string|uppercase-string` is weird.

I recommend to write `assertType('uppercase-string', $value);` and we'll need to debug this.

Reply:
Addressed in e99bf4f85